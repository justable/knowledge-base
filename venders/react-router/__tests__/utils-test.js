import React from 'react'
import {
  Routes,
  Route,
  createRoutesFromChildren,
  matchRoutes
} from '../../react-router'
import type { RouteObject } from '../types'

describe('utils', () => {
  beforeEach(() => {})

  afterEach(() => {})

  describe('createRoutesFromChildren', () => {
    let routes: RouteObject[] = []
    function pickPaths(routes: RouteObject[], pathname: string) {
      let matches = matchRoutes(routes, { pathname })
      return matches ? matches.map((match) => match.route.path) : null
    }
    function Admin() {
      return <h1>Admin</h1>
    }
    function Manage() {
      return <h1>Manage</h1>
    }
    const MockRoutes = (basename = '') => {
      return (
        <Routes basename={basename}>
          <Route path="/admin" element={<Admin />} />
          <Route path="/manage" element={<Manage />} />
        </Routes>
      )
    }
    it('把ReactElement转换成树状结构', () => {
      routes = createRoutesFromChildren(MockRoutes().props.children)
      expect(routes).toMatchSnapshot()
    })
    it('匹配当前路由的children', () => {
      expect(pickPaths(routes, '/admin')).toEqual(['/admin'])
    })
  })
})
