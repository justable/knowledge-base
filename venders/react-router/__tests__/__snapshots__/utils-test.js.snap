// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils createRoutesFromChildren 把ReactElement转换成树状结构 1`] = `
Array [
  Object {
    "caseSensitive": false,
    "element": <Route
      element={<Admin />}
      path="/admin"
    />,
    "path": "/admin",
  },
  Object {
    "caseSensitive": false,
    "element": <Route
      element={<Manage />}
      path="/manage"
    />,
    "path": "/manage",
  },
]
`;
